test/test_oneagentmetadataenricher.py

```diff
@@ -58,10 +75,122 @@ def test_mock_get_metadata_file(self, mock_func):
     @patch('dynatrace.opentelemetry.metrics.export.oneagentmetadataenricher'
            '.OneAgentMetadataEnricher._parse_oneagent_metadata')
     def test_tags_overwritten(self, mock_func):
-        enricher = OneAgentMetadataEnricher(logging.Logger(__name__))
+        enricher = OneAgentMetadataEnricher()
         tags = {"tag1": "value1"}
         mock_func.return_value = {"tag1": "newValue"}
 
         enricher.add_oneagent_metadata_to_tags(tags)
 
         self.assertEqual(tags, {"tag1": "newValue"})
+
+    @patch('dynatrace.opentelemetry.metrics.export.oneagentmetadataenricher'
+           '.OneAgentMetadataEnricher._get_metadata_file_content')
+    def test_parse_valid_data(self, mock_func):
+        mock_func.return_value = ["k1=v1\n", "k2=v2"]
+
+        enricher = OneAgentMetadataEnricher()
+        res = enricher._parse_oneagent_metadata(
+            enricher._get_metadata_file_content())
```

Comment:
Since you pass the values anyway you could omit mocking the method and just pass it right away.

Reply:
:sweat_smile: